### PR TITLE
Update requirements.txt

### PR DESCRIPTION
**Pull request by maurizi**
Alphabetizes listed requirements

Updates:
 - boto
 - Django
 - django-apptemplates
 - django-registration-redux
 - django-storages
 - gunicorn
 - jsonschema
 - Pillow
 - psycopg2
 - python-dateutil
 - pytz

Adds transitive dependencies to requirements.txt

Depends on https://github.com/OpenTreeMap/otm-cloud/pull/406

Connects to #3113
Connects to #3114
Connects to #1812

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,38 +1,55 @@
...
-Django==1.11.3  # rq.filter: >=1.11,<1.12
-django-apptemplates==1.2
+amqp==1.4.9
+anyjson==0.3.3
+billiard==3.3.0.23
+boto==2.48.0                             # http://docs.pythonboto.org/en/latest/releasenotes/v2.39.0.html
 # django-celery requires celery, but is incompatible with celery
 # 4.0. django-celery has added this restriction on the master branch,
 # but it has not yet been released.
 celery>=3.1.15,<4.0
```
Comment (jwalgran) [inline requirements.txt]: Our celery is still "floating" in this requirements file. Are we not pinning that to a specific version by design?

Comment (maurizi) [inline requirements.txt]: Fixed in (rebased) commit https://github.com/OpenTreeMap/otm-core/pull/3184/commits/a971d4312fe45b5e44cd05986f6e777385cf8d50

File: requirements.txt [hunk 2]
```diff
@@ -1,38 +1,55 @@
 # For outdated module report see https://requires.io/github/OpenTreeMap/otm-core/requirements/?branch=develop
 # Changelog URL shown below if not linked on above page
 
-boto==2.42.0                             # http://docs.pythonboto.org/en/latest/releasenotes/v2.39.0.html
-# https://docs.djangoproject.com/en/1.10/releases/#id2
-Django==1.11.3  # rq.filter: >=1.11,<1.12
-django-apptemplates==1.2
+amqp==1.4.9
+anyjson==0.3.3
+billiard==3.3.0.23
```
Comment (jwalgran) [inline requirements.txt]: On the develop branch I can export a CSV but on this branch I get these errors. Not sure if it is actually a problem with billiard.

```
[2017-08-10 12:52:03,957: INFO/MainProcess] Received task: exporter.tasks.async_csv_export[242bf504-3aaa-46d9-a915-fd6096a9b4f5]
[2017-08-10 12:52:03,965: DEBUG/MainProcess] Task accepted: exporter.tasks.async_csv_export[242bf504-3aaa-46d9-a915-fd6096a9b4f5] pid:11152
[2017-08-10 12:52:07,646: ERROR/MainProcess] Task exporter.tasks.async_csv_export[242bf504-3aaa-46d9-a915-fd6096a9b4f5] raised unexpected: WorkerLostError('Worker exited prematurely: signal 11 (SIGSEGV).',)
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/billiard/pool.py", line 1175, in mark_as_worker_lost
    human_status(exitcode)),
WorkerLostError: Worker exited prematurely: signal 11 (SIGSEGV).
```

Comment (maurizi) [inline requirements.txt]: This should be fixed by https://github.com/OpenTreeMap/otm-core/pull/3184/commits/1147a7e3a8ad5d4f60a7f4b5e5c80857795f7df2

File: opentreemap/importer/tasks.py [hunk 3]
```diff
@@ -67,7 +67,7 @@ def _create_rows(ie, reader):
         RowModel.objects.bulk_create(rows)  # create final partial block
 
 
-@task()
+@shared_task()
 def run_import_event_validation(import_type, import_event_id, file_obj):
```
Comment (jwalgran) [inline opentreemap/importer/tasks.py]: If an import file has fewer rows than our row chunk size (85) no import rows were being created. It turns out that an exception was being thrown due to the fact that celery 4 now converts a chord with only one task in the head into a simple AsyncResult, which does not have a savable group result parent (http://docs.celeryproject.org/en/latest/whatsnew-4.0.html)

Here is a test file with one species that will work for an instance centered on LA that works in staging but not on this branch [la_test_species_import.csv.zip](https://github.com/OpenTreeMap/otm-core/files/1232322/la_test_species_import.csv.zip)

Wrapping the group result saving in a conditional fixed the issue for me

```diff
diff --git a/opentreemap/importer/tasks.py b/opentreemap/importer/tasks.py
index 2702940c..9b0c0e5f 100644
--- a/opentreemap/importer/tasks.py
+++ b/opentreemap/importer/tasks.py
@@ -6,6 +6,7 @@ from __future__ import division
 import json
 
 from celery import shared_task, chord
+from celery.result import GroupResult
 from django.core.exceptions import ObjectDoesNotExist
 from django.conf import settings
 from django.db import transaction
@@ -99,10 +100,14 @@ def run_import_event_validation(import_type, import_event_id, file_obj):
         final_task = _finalize_validation.si(import_type, import_event_id)
 
         async_result = chord(validation_tasks, final_task).delay()
-        group_result = async_result.parent
-        if group_result:  # Has value None when run in unit tests
-            group_result.save()
-            ie.task_id = group_result.id
+        async_result_parent = async_result.parent
+        if async_result_parent:  # Has value None when run in unit tests
+            if isinstance(async_result_parent, GroupResult):
+                # Celery 4 converts a chord with only one task in the head into
+                # a simple chain, which does not have a savable parent
+                # GroupResult.
+                async_result_parent.save()
+            ie.task_id = async_result_parent.id
 
         _assure_status_is_at_least_verifying(ie)
 

```

File: opentreemap/opentreemap/__init__.py [hunk 4]
```diff
@@ -0,0 +1,5 @@
+from __future__ import absolute_import, unicode_literals
```
Comment (maurizi) [inline opentreemap/opentreemap/__init__.py]: Yes. This is from the Celery guide on Django integration: http://docs.celeryproject.org/en/latest/django/first-steps-with-django.html

**Pull request merged**